lib/src/async/concat.dart

```diff
@@ -71,7 +71,7 @@ class _ConcatStream<T> extends Stream<T> {
     }, onResume: () {
       if (currentSubscription != null) currentSubscription.resume();
     }, onCancel: () {
-      if (currentSubscription != null) return currentSubscription.cancel();
```

Comment:
Re-reading the `onCancel` docs I think we should keep returning. A `return null` after this line should fix the error.

Reply:
Done.